src/Application/NBB.Application.DataContracts/Command.cs

```diff
@@ -1,10 +1,9 @@
 ﻿using MediatR;
 using NBB.Core.Abstractions;
-using System;
 
 namespace NBB.Application.DataContracts
 {
-    public abstract class Command : ICommand, IMetadataProvider<CommandMetadata>
+    public abstract class Command : ICommand, IRequest, IMetadataProvider<CommandMetadata>
     {
         public CommandMetadata Metadata { get; }

```

Comment:
Generic command should not derive from Command. It should implement ICommand instead.
Also should not implement IRequest, but IRequest<TResponse>